### PR TITLE
[DEVOPS-6292] Migrate CICD from CircleCI to Github Actions

### PR DESCRIPTION
**Pull request by anaharris**
Jira: https://codecademy.atlassian.net/browse/DEVOPS-6292

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        path: vendor/bundle
+        key: ${{ runner.os }}-bundle-{{ hashFiles("**/Gemfile.lock") }}
+        restore-keys: |
+          ${{ runner.os }}-${{ hashFiles('**/Gemfile.lock') }}
+          ${{ runner.os }}-${{ github.ref_name }}
+          ${{ runner.os }}-main
+          ${{ runner.os }}-
+  
+    - name: Run Rubocop
+      run: bundle exec rubocop --config ./.rubocop.yml
```
Comment (dantech2000) [inline .github/workflows/lint.yaml]: LGTM only thing is the no new line at the end of the file but that might just be a yaml linting thing

**Review by dantech2000 — APPROVED** [review]
LGTM 👍🏼

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Setup Ruby 
+      uses: ruby/setup-ruby@v1 # no need for ruby-version because we have a .ruby_version file
+      with:
+          bundler-cache: true
+
+    - name: Cache dependencies
```
Comment (mbacchi) [inline .github/workflows/lint.yaml]: I wonder if it's even necessary to cache dependencies for a project like this which is built very rarely. (I'm also not sure how long the cache stays in place, and if it's purged automatically or if it costs money to maintain it in Github Actions' cache.)

I'll leave it up to you to decide Ana.

**Review by mbacchi — APPROVED** [review]
Looks good, one minor comment but that shouldn't block you.

**Review by leonpham0 — APPROVED** [review]

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (anaharris) [reply .github/workflows/lint.yaml]: I agree, we probably don't need caching on this repo. No one has touched it in ages and afaik there aren't any updates to it happening soon. We can revisit but for now I'll remove caching.

**Pull request merged**